vint/linting/cli.py

```diff
@@ -116,11 +117,17 @@ def _lint_all(self, env, config_dict):
         violations = []
         linter = self._build_linter(config_dict)
 
-        for file_path in paths_to_lint:
-            violations += linter.lint_file(file_path)
+        if self._should_read_from_stdin(env):
+            violations += linter.lint_text(sys.stdin.read())
+        else:
+            for file_path in paths_to_lint:
+                violations += linter.lint_file(file_path)
 
         return violations
 
+    def _should_read_from_stdin(self, env):
+        return len(env['file_paths']) == 1 and PosixPath('-') in env['file_paths']
```

Comment:
Couldn't this be `env['file_paths'] == [PosixPath('-')]`?

But then again you might even want to mix it - `vint foo.vim - bar.vim` could read 3 "files".

Reply:
Agree. I feel supporting `vint foo.vim - bar.vim` is a good way to follow common usage of `-` such as `cat a - b`.